cfme/tests/infrastructure/test_host.py

```diff
@@ -40,7 +55,38 @@ def navigate_and_select_quads(provider):
     return edit_view
 
 
+@pytest.mark.uncollectif(lambda provider: not provider.one_of(VMwareProvider))
+def test_discover_host(request, provider, appliance, hosts_ipaddresses):
+    if provider.delete_if_exists(cancel=False):
+        provider.wait_for_delete()
+
+    collection = appliance.collections.hosts
+
+    def _cleanup():
+        all_hosts = collection.all(provider)
+        if all_hosts:
+            collection.delete(*all_hosts)
+
+    _cleanup()
+    request.addfinalizer(_cleanup)
+
+    first_addr = provider.ip_address if hasattr(provider, 'ip_address') else hosts_ipaddresses[0]
```

Comment:
REQUIRED: discover expects you to provide range of ips where. according to your code you expect that provider and hosts are in the same network/sub network. I doubt this assumption is right. I'd probaby take host ip and made a subnet which would include host ip.

Reply:
Hmm, I'll investigate this more. Also some providers have ``discovery_range`` that I probably want to use when available.